extensions/ql-vscode/src/remote-queries/gh-actions-api-client.ts

```diff
@@ -300,20 +298,16 @@ async function downloadArtifact(
     archive_format: 'zip',
   });
   const artifactPath = path.join(tmpDir.name, `${artifactId}`);
-  await saveFile(`${artifactPath}.zip`, response.data as ArrayBuffer);
-  await unzipFile(`${artifactPath}.zip`, artifactPath);
+  await unzipBuffer(response.data as ArrayBuffer, `${artifactPath}.zip`, artifactPath);
   return artifactPath;
 }
 
-async function saveFile(filePath: string, data: ArrayBuffer): Promise<void> {
+async function unzipBuffer(data: ArrayBuffer, filePath: string, destinationPath: string): Promise<void> {
```

Comment:
Would it make sense to also move this function to the `pure/zip` directory, or do you think this will only be used in this file?

Reply:
It seems very specific so I thought i'd leave it there. It's also likely going away with live results as will most of this file!